packages/core/src/renderer/components/+namespaces/hierarchical-namespaces.injectable.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import namespaceStoreInjectable from "./store.injectable";
+
+const hierarchicalNamespacesInjectable = getInjectable({
+  id: "hierarchical-namespaces",
+
+  instantiate: (di) => {
+    const namespaceStore = di.inject(namespaceStoreInjectable);
+
+    return namespaceStore.getByLabel(["hnc.x-k8s.io/included-namespace=true"]);
```

Comment:
Probably we should keep such label / api-related data as `hnc.x-k8s.io/included-namespace=true` closer to api definitions. So we don't need to create new injectable each time just for single method.
So this could be even just a method of `NamespaceStore.getNamespacesTree(optionalParamsForLabel = {})` or similar. Wdyt?

Reply:
Refactored the PR, moved all tree build logic to store and namespace API.